src/Processing/DefaultProcessor.php

```diff
@@ -395,17 +399,20 @@ protected function handleMissingFields(
 						? $this->process([], $structureArgs->type, $options)
 						: $this->processWithoutInitialization([], $structureArgs->type, $options);
 				} catch (InvalidData $exception) {
-					$type->overwriteInvalidField($missingField, $exception->getInvalidType());
+					$type->overwriteInvalidField($missingField, $exception);
 				}
 			} elseif ($requiredFields !== $options::REQUIRE_NONE && !$type->isFieldInvalid($missingField)) {
 				// Field is missing and have no default value, mark as invalid
 				$propertyRuleMeta = $propertyMeta->getRule();
 				$propertyRule = $this->ruleManager->getRule($propertyRuleMeta->getType());
 				$type->overwriteInvalidField(
 					$missingField,
-					$propertyRule->createType(
-						$this->createRuleArgsInst($propertyRule, $propertyRuleMeta),
-						$this->getTypeContext(),
+					ValueDoesNotMatch::create(
+						$propertyRule->createType(
+							$this->createRuleArgsInst($propertyRule, $propertyRuleMeta),
+							$this->getTypeContext(),
+						),
+						$data,
```

Comment:
`$data` is for whole VO, nut just for one field. This is the case for `new NoValue`